primitives/src/traits/dispute_api.rs

```diff
@@ -61,20 +57,30 @@ pub trait DisputeApi {
     ///
     /// Returns the dispute mechanism's report if available, otherwise `None`. If `None` is
     /// returned, this means that the dispute could not be resolved.
-    fn on_resolution(
-        disputes: &[MarketDispute<Self::AccountId, Self::BlockNumber>],
+    fn get_resolution_outcome(
         market_id: &Self::MarketId,
         market: &MarketOfDisputeApi<Self>,
     ) -> Result<Option<OutcomeReport>, DispatchError>;
 
+    /// Allow the flow of funds to the market dispute mechanism.
+    /// **May** assume that `market.dispute_mechanism` refers to the calling dispute API.
+    ///
+    /// # Returns
+    /// Returns the negative imbalance which is meant to be used for the treasury.
+    fn maybe_pay(
+        market_id: &Self::MarketId,
+        market: &MarketOfDisputeApi<Self>,
+        resolved_outcome: &OutcomeReport,
```

Comment:
this is a good one. because `get_resolution_outcome` allows the flexibility to return `None`, the MDM is not required to know the default option of the prediction markets pallet (which is the report outcome). Indeed, the MDM could just query the market internally and fetch the reported outcome, but in our previous design this was not required.

So we know the resolution outcome of the MDM, but additionally what happens if a global dispute was triggered? Then we would like to override the resolution outcome of the MDM and make payment decisions based on the final resolved outcome (which could be decided by the global dispute pallet).

Reply:
By the time `exchange` is called the market's `resolved_outcome` was not already written.